src/prometheus.rs

```diff
@@ -0,0 +1,189 @@
+use std::{env, error::Error};
+
+use prometheus::register_gauge;
+use rustic_core::repofile::SnapshotFile;
+
+pub fn parse_label(
+    s: &str,
+) -> Result<(String, String), Box<dyn Error + Send + Sync + 'static>> {
+    let pos = s
+        .find('=')
+        .ok_or_else(|| format!("invalid prometheus label definition: no `=` found in `{s}`"))?;
+    Ok((s[..pos].to_owned(), s[pos + 1..].to_owned()))
+}
+
+pub fn publish_metrics(
+    pushgateway_url: &str,
+    job_name: &str,
+    snap: &SnapshotFile,
+    user_labels: Vec<(String, String)>,
+) -> anyhow::Result<()> {
+    let summary = snap.summary.as_ref().expect("Reaching the 'push to prometheus' point should only happen for successful backups, which must have a summary set.");
+
+    let metric_time = register_gauge!("rustic_backup_time", "Timestamp of this snapshot",).unwrap();
+    metric_time.set(snap.time.timestamp_millis() as f64 / 1000.);
+    let metric_files_new = register_gauge!(
+        "rustic_backup_files_new",
+        "New files compared to the last (i.e. parent) snapshot",
+    )
+    .unwrap();
```

Comment:
I think we can just bubble up all errors from the `register_gauge!()` by using `?`.

Reply:
Good point! I had done it this way because for some reason prometheus-using people always do that. I also added `.context()` just in case :)